src/RelayClient.ts

```diff
@@ -44,16 +76,206 @@ class RelayClient extends EnvelopingEventEmitter {
     this._envelopingConfig = getEnvelopingConfig();
   }
 
-  public _getRelayRequestDetails = async (
-    // FIXME: make private
-    _request: UserDefinedRelayRequest
-  ): Promise<RelayRequest> => {
-    return await Promise.resolve({} as EnvelopingTypes.RelayRequestStruct);
+  private estimateTokenTransferGas = async (
+    _props: TokenGasEstimationParams
+  ): Promise<BigNumber> => {
+    return Promise.resolve(constants.Zero);
+  };
+
+  public estimateInternalCallGas = async (
+    _params: EstimateInternalGasParams
+  ): Promise<BigNumber> => {
+    return Promise.resolve(constants.Zero);
   };
 
-  // TODO: lift the ts exception once method is used
   // eslint-disable-next-line @typescript-eslint/ban-ts-comment
-  // @ts-ignore
+  // @ts-ignore FIXME: remove the ts ignore once the method is consumed
+  private _getEnvelopingRequestDetails = async (
+    envelopingRequest: UserDefinedEnvelopingRequest,
+    {
+      isSmartWalletDeploy,
+      preDeploySWAddress,
+      forceGasLimit,
+      forceGasPrice,
+      internalEstimationCorrection,
+      addCorrection,
+    }: RequestConfig
+  ): Promise<EnvelopingRequest> => {
+    const isDeployment: boolean = isDeployRequest(
+      envelopingRequest as EnvelopingRequest
+    );
+    const {
+      relayData: { callForwarder },
+      request: { data, from, to, tokenContract },
+    } = envelopingRequest;
+
+    if (!callForwarder) {
+      throw new Error(MISSING_CALL_FORWARDER);
+    }
+
+    const callVerifier: PromiseOrValue<string> =
+      envelopingRequest.relayData.callVerifier ||
+      this._envelopingConfig[
+        isDeployment ? 'deployVerifierAddress' : 'relayVerifierAddress'
+      ];
+
+    if (!callVerifier) {
+      throw new Error('No call verifier present. Check your configuration.');
+    }
+
+    const relayServerUrl = this._envelopingConfig.preferredRelays.at(0)?.url;
```

Comment:
During the relayTransaction, we ping/query the servers and try to execute the transaction using their services, Im guessing that the feesReceiver can be retrieve during that process, instead of get from the first server available.

Reply:
would that happen prior to calling this method then?

Reply:
yes the way that I see it is relayTransaction -> selectNextRelay -> _getEnvelopingRequestDetails -> _prepareRelayHttpRequest.

But Im not pretty sure but when the selectNextRelay is called we will have a server available that give us the feesReceiver, since its the one that will process the transaction, this needs to be different for each server, he question part that I still does not have clear if we should do the _getEnvelopingRequestDetails and _prepareRelayHttpRequest for each request or change just the necessary.

Reply:
changed